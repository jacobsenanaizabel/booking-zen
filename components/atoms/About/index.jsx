import React from 'react';
import PropTypes from 'prop-types';
//import { Container, ButtonStyled } from './styles';

const About = () => {

  return (
    <></>
  );
};

About.propTypes = {
  errorHandler: PropTypes.func,
};

About.defaultProps = {
  errorHandler: null,
};
export default About;
