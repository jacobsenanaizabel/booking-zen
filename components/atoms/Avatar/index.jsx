import React from 'react';
import PropTypes from 'prop-types';
import { ContainerAvatar, User } from './styles';

const Avatar = () => {

  return (
    <>
      <ContainerAvatar>
        <figure>
          <User src="https://zenklub.com.br/static/professionals/5ec432ab26f9535a51a26507/445fdbfce83fd6de0d7f057bc52af9da23959f86-180x180.png"/>
        </figure>
      </ContainerAvatar>
    </>
  );
};

Avatar.propTypes = {
  errorHandler: PropTypes.func,
};

Avatar.defaultProps = {
  errorHandler: null,
};
export default Avatar;
