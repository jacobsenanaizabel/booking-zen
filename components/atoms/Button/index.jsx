import React from 'react';
import PropTypes from 'prop-types';
import { Container, ButtonStyled } from './styles';

const Button = ({text, ...other}) => {

  return (
    <Container>
      <ButtonStyled
        tabindex="0"
        role="button"
        {...other}
      >
       {text}
      </ButtonStyled>
    </Container>
  );
};

Button.propTypes = {
  text: PropTypes.string,
};

Button.defaultProps = {
  text: 'OK'
};
export default Button;
