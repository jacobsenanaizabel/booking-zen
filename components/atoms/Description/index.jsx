import React from 'react';
import PropTypes from 'prop-types';
import { ContainerDescription } from './styles';
import {Text} from '../index'

const Description = ({text}) => {

  return (
    <>
    <ContainerDescription>
      <Text text={text} color="text" size="1rem"/>
    </ContainerDescription>
    </>
  );
};

Description.propTypes = {
  errorHandler: PropTypes.func,
};

Description.defaultProps = {
  errorHandler: null,
};
export default Description;
