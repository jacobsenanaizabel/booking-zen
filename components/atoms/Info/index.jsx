import React from 'react';

import PropTypes from 'prop-types';
import { Text } from '../index';
import { Info as About, ContainerInfo } from './styles';

const Info = ({location, profession}) => {   
  return (
    <>
    <ContainerInfo>
    <About>
      <Text text={profession} size="1rem" color="black" weight="700"/> 
      <Text text={'- '+location} size="1rem" color="purple" weight="700"/>
    </About>
  </ContainerInfo>
    </>
  );
};


Info.propTypes = {
  location: PropTypes.string,
  profession: PropTypes.string
};

export default Info;
