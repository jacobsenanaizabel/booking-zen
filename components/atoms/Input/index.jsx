import React from 'react';
import { InputStyled } from './styles';

const Input = ({...other}) => {
  return (
    <>
      <InputStyled {...other} data-testid="input"/>
    </>
  );
}

Input.propTypes = {
};

Input.defaultProps = {
};

export default Input;


