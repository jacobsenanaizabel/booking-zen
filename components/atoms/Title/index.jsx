import React from 'react';
import PropTypes from 'prop-types';

import {
  Heading, 
  ContainerTitle
} from './styles';

const Title = ({text}) => {
  return (
    <>
    <ContainerTitle>
        <Heading>{text}</Heading>
    </ContainerTitle>
    </>
  );
};

Title.propTypes = {
  text: PropTypes.string
};


export default Title;
