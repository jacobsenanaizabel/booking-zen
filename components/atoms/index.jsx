export { default as Title } from './Title';
export { default as Input } from './Input';
export { default as Button } from './Button';
export { default as About } from './About';
export { default as Avatar } from './Avatar';
export { default as Description } from './Description';
export { default as Price } from './Price';
export { default as Rating } from './Rating';
export { default as Text } from './Text';