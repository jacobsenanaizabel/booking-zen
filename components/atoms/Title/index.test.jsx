import React from 'react';
import { render } from "@testing-library/react";
import {Title} from '@components/atoms/index';

describe('Title tests', () => {

  const fakeText = "Testing text";

  it('should match snapshot with default Title', () => {

    const { asFragment } = render(<Title />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Title with text', () => {


    const { asFragment } = render(<Title text={fakeText} />);
    expect(asFragment()).toMatchSnapshot();
  });

});