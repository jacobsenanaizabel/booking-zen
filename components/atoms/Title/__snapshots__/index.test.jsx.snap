// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title tests should match snapshot with default Title 1`] = `
<DocumentFragment>
  <div
    class="styles__ContainerTitle-sc-17f0diz-1 kMIxue"
  >
    <h1
      class="styles__Heading-sc-17f0diz-0 iYJgeC"
    />
  </div>
</DocumentFragment>
`;

exports[`Title tests should match snapshot with default Title with text 1`] = `
<DocumentFragment>
  <div
    class="styles__ContainerTitle-sc-17f0diz-1 kMIxue"
  >
    <h1
      class="styles__Heading-sc-17f0diz-0 iYJgeC"
    >
      Testing text
    </h1>
  </div>
</DocumentFragment>
`;
